client-tools/Import/ImportHelper.cpp

```diff
@@ -590,6 +599,253 @@ bool ImportHelper::importJson(std::string const& collectionName,
   return !_hasError;
 }
 
+bool ImportHelper::importJsonWithRewrite(std::string const& collectionName,
+                                         std::string const& pathName,
+                                         bool assumeLinewise) {
+  /**************
+   * This Block is copied over from importJson
+   * If this test works out we should unify them
+   ***************/
+  ManagedDirectory directory(_encryption, TRI_Dirname(pathName), false, false,
+                             true);
+  std::string fileName(TRI_Basename(pathName.c_str()));
+  _collectionName = collectionName;
+  _firstLine = "";
+  _outputBuffer.clear();
+  _errorMessages.clear();
+  _hasError = false;
+
+  if (!checkCreateCollection()) {
+    return false;
+  }
+  if (!collectionExists()) {
+    return false;
+  }
+
+  // read and convert
+  int64_t totalLength;
+  std::unique_ptr<arangodb::ManagedDirectory::File> fd;
+
+  if (fileName == "-") {
+    // we don't have a filesize
+    totalLength = 0;
+    fd = directory.readableFile(STDIN_FILENO);
```

Comment:
Does this work on Windows?

Reply:
I hope so, test will show.
Anyways this is taken from the other import mechanism, if it does not work we need a separate bugfix for this in general

Reply:
I'm just surprised because I remember some ifdefs to exempt Windows from using stdin/stout in some client-tool code, but it may have changed since or it was always limited to just a few corner cases. I found the following:
- Handling of Windows' UTF-16 encoding
  https://github.com/arangodb/arangodb/blob/devel/client-tools/Shell/ShellConsoleFeature.cpp#L335-L344
- Using pipes is Linux-only in the VPack util
  https://github.com/arangodb/arangodb/blob/devel/client-tools/VPack/VPackFeature.cpp#L134-L147